Assets/Scripts/KeyboardInput.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace movement
+{
+
+    public class KeyboardInput : MonoBehaviour
+    {
+        // Update is called once per frame
+        void Update()
+        {
+            if (Input.GetKey(KeyCode.D))
```

Comment:
I think this entire code structure can be improved since you're setting something to true if a conditional is true and false otherwise so maybe something like:

```suggestion
            VirtualInputManager.Instance.MoveRight = Input.GetKey(KeyCode.D);
```

Reply:
I'm not sure I understand what you're suggesting